core/dbt/parser/generic_test_builders.py

```diff
@@ -182,6 +157,45 @@ def __init__(
             if short_name != full_name and "alias" not in self.config:
                 self.config["alias"] = short_name
 
+    def _process_legacy_args(self):
+        config = {}
+        for key in self.CONFIG_ARGS:
+            value = self.args.pop(key, None)
+            if value and "config" in self.args and key in self.args["config"]:
+                raise SameKeyNestedError()
+            if not value and "config" in self.args:
+                value = self.args["config"].pop(key, None)
+            if value is not None:
```

Comment:
This check is also happening in `self._render_values`. I think we can always update the dict with this key/value pair.